requests_ntlm/requests_ntlm.py

```diff
@@ -36,33 +35,25 @@ def __init__(self, username, password, session=None):
 
         self.password = password
 
-        self.adapter = HTTPAdapter()
-
-        # Keep a weak reference to the Session, if one is in use.
-        # This is to avoid a circular reference.
-        self.session = weakref.ref(session) if session else None
-
     def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
                                    response, args):
         """Attempt to authenticate using HTTP NTLM challenge/response."""
         if auth_header in response.request.headers:
             return response
 
-        request = copy_request(response.request)
-
-        content_length = int(request.headers.get('Content-Length', '0'),
+        content_length = int(response.request.headers.get('Content-Length', '0'),
                              base=10)
-        if hasattr(request.body, 'seek'):
+        if hasattr(response.request.body, 'seek'):
             if content_length > 0:
-                request.body.seek(-content_length, 1)
+                response.request.body.seek(-content_length, 1)
             else:
-                request.body.seek(0, 0)
+                response.request.body.seek(0, 0)
 
-        adapter = self.adapter
-        if self.session:
-            session = self.session()
-            if session:
-                adapter = session.get_adapter(response.request.url)
+        # Consume content and release the original connection
+        # to allow our new request to reuse the same one.
+        response.content
+        response.raw.release_conn()
```

Comment:
I *think* this logic is insufficient. It works fine in the single-threaded case (which might be good enough), but urllib3's connection-pooling logic will mean that releasing the connection in this way could lead to a failure to get it back.

@sigmavirus24, is it worth trying to explicitly hold on to the connection here?